### DISCUSSION TITLE
Transitioning from quarkus JVM to GraalVM

### DISCUSSION THREAD

**Discussion by UrCoderPal** [quarkusio/quarkus#42973] [Q&A]
used tips and solution from issue #34854 but now stuck with a problem: 
I want to substitute the org.jose4j.keys.EdDsaKeyUtil class which extends the OctetKeyPairUtil and this the KeyPairUtil class which is of 'default' access.
So now when i have to override the abstract methods of KeyPairUtil in my TargetEdDsaKeyUtil class i am restricted. I thought of creating substitutes for all three but then in my TargetKeyPairUtil the @TargetClass(org.jose4j.keys.KeyPairUtil.class) also gives an error because the value we are passing here is inaccessible.
I intended to keep all the implementation logic intact.
Selectively substituting methods causing build issues is an option but in my case those methods are using default access methods of their superclass so then replicating each such method and fields(using @Alias annotation) could later on cause me logical issues and may break the code eventually.
What do u guys suggest?

![TargetClassBuildIssue](https://github.com/user-attachments/assets/d72f4d42-76a1-4f7d-b1cc-7d8f15a39e9a)

@gsmet @zakkak @geoand

**Comment by zakkak** [comment]
For classes that are not under your control (coming from 3rd party libraries) I am afraid you can't do much. For classes under your control you can either refactor them to be _native-image friendly_ or use helper functions that reference the super methods and fields to avoid the need of extensive `@alias`.

> in my TargetKeyPairUtil the @TargetClass(org.jose4j.keys.KeyPairUtil.class) also gives an error because the value we are passing here is inaccessible

Do you mean that `org.jose4j.keys.KeyPairUtil` is not accessible if so you can just use its name, e.g.

```
@TargetClass(className = "org.jose4j.keys.KeyPairUtil")
```

**Reply by gsmet** [reply]
That being said, we have Jose4j in the classpath so maybe we would need more details on if it's a use case we need to support.

What exactly are you trying to do?
